app/src/main/java/ru/gelin/android/weather/notification/ParcelableWeather2.java

```diff
@@ -67,22 +79,105 @@ private ParcelableWeather2(Parcel in) {
+                    main.put("humidity", forecastBundle.getInt("weather_humidity_value"));
+                    main.put("temp_min", forecastLowTemp);
+                    main.put("temp_max", forecastHighTemp);
+
+                    //forecast
+                    //"city":{"id":3181913,"name":"Bolzano","coord":{"lat":46.4927,"lon":11.3336},"country":"IT"}
+                    city.put("name", location);
+                    city.put("country", "World");
+                    reconstructedForecast.put("city", city);
+
+                    item.put("dt", (time / 1000) + timeOffset);
+                    item.put("main", main);
+                    item.put("weather", weather);
+                    list.put(item);
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
+
+                // get the rest
+                while (--conditions > 0) {
+                    conditionBundle = in.readBundle();
+                    conditionBundle.getString("weather_condition_text");
+                    weatherConditionTypesToOpenWeatherMapIds(conditionBundle.getStringArray("weather_condition_types")[0]);
+                    conditionBundle.getInt("weather_current_temp");
+                    item = new JSONObject();
+                    condition = new JSONObject();
+                    main = new JSONObject();
+                    weather = new JSONArray();
+                    timeOffset += 86400000;
+                    try {
+                        condition.put("id", weatherConditionTypesToOpenWeatherMapIds(conditionBundle.getStringArray("weather_condition_types")[0]));
+                        condition.put("main", conditionBundle.getString("weather_condition_text"));
+                        condition.put("icon", Weather.mapToOpenWeatherMapIcon(weatherConditionTypesToOpenWeatherMapIds(conditionBundle.getStringArray("weather_condition_types")[0])));
+                        weather.put(condition);
+
+                        main.put("temp", conditionBundle.getInt("weather_current_temp"));
+                        main.put("humidity", conditionBundle.getInt("weather_humidity_value"));
+                        main.put("temp_min", conditionBundle.getInt("weather_low_temp"));
+                        main.put("temp_max", conditionBundle.getInt("weather_high_temp"));
+
+                        item.put("dt", (time / 1000) + timeOffset);
+                        item.put("main", main);
+                        item.put("weather", weather);
+                        list.put(item);
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+                }
+                try {
+                    reconstructedForecast.put("cnt", list.length());
+                    reconstructedForecast.put("list", list);
+
+                } catch (JSONException e) {
+                    e.printStackTrace();
```

Comment:
I guess if we get any of the exceptions here, we should abort the whole thing, since it's unlikely it would work correctly, right?

Reply:
Not really, the weather is also used in other forms (not JSON-based)